### PR TITLE
[16.0][MIG] mail_outbound_static: Migration to 16.0

### PR DESCRIPTION
**Pull request by dreispt**

### CODE REVIEW

File: mail_outbound_static/models/ir_mail_server.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            bounce_alias = (
+                self.env["ir.config_parameter"].sudo().get_param("mail.bounce.alias")
+            )
+            if not bounce_alias:
+                # then, bounce handling is disabled and we want
+                # Return-Path = From
+                if "Return-Path" in message:
+                    message.replace_header("Return-Path", email_from)
+                else:
+                    message.add_header("Return-Path", email_from)
```
Comment (yostashiro) [inline mail_outbound_static/models/ir_mail_server.py:113]: Update of smtp_from and message values should be done by extending _prepare_email_message(), or the updated values get overriden in there.  I suppose the same issue exists in 15.0.

**Review by yostashiro — CHANGES_REQUESTED** [review]
@dreispt Thanks for the PR.  Can you please check this commit https://github.com/qrtl/social/commit/196f6b3499f3c55aa6da326cea958c2deb2be632 which I added on top of yours, and cherry-pick this as appropriate?

**Review by CasVissers-360ERP — APPROVED** [review]
We have it running for some months now, all good.

**Comment by adrienpeiffer** [comment]
Hi @dreispt 
I have an issue related with the smtp_sesssion which is set here 

https://github.com/odoo/odoo/blob/439bef710b78afe0e5ba1d54ce04a5e5c3906cdc/addons/mail/models/mail_mail.py#L397

The content of the FROM is replaced by the value on the session here :

https://github.com/odoo/odoo/blob/15aa0e5025e35a5093bd70cae0513cb492806dcc/odoo/addons/base/models/ir_mail_server.py#L602 

Do you have this kind of issue ?

**Comment by yostashiro** [comment]
@adrienpeiffer I believe https://github.com/OCA/social/pull/1051 will fix the issue you mention.

File: mail_outbound_static/models/ir_mail_server.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            domain_whitelist = self._get_domain_whitelist(item.domain_whitelist)
+            if email_domain in domain_whitelist:
+                mail_server_id = item.id
+                break
+        if not mail_server_id:
+            mail_server_id = self.sudo().search([], order="sequence", limit=1).id
+        return mail_server_id
+
+    @api.model
+    def create(self, values):
```
Comment (adrienpeiffer) [inline mail_outbound_static/models/ir_mail_server.py]: You should use 
```
@api.model_create_multi
def create(self, vals_list):
```

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]

File: mail_outbound_static/models/ir_mail_server.py (hunk 2 above)
Comment (dreispt) [reply mail_outbound_static/models/ir_mail_server.py]: Done

File: mail_outbound_static/models/ir_mail_server.py (hunk 1 above)
Comment (dreispt) [reply mail_outbound_static/models/ir_mail_server.py:113]: @yostashiro It is best if you forward port #1051 to 16.0

**Comment by dreispt** [comment]
Fixed and rebased.
Please re-review.

**Review by hbrunn — APPROVED** [review]

**Pull request merged**